src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java

```diff
@@ -182,7 +184,12 @@ public void initFlowFrameworkIndexIfAbsent(FlowFrameworkIndex index, ActionListe
                                                     }
                                                 }, exception -> {
                                                     logger.error("Failed to update index setting for: " + indexName, exception);
-                                                    internalListener.onFailure(exception);
+                                                    internalListener.onFailure(
+                                                        new FlowFrameworkException(
+                                                            exception.getMessage(),
```

Comment:
Should we give more detailed messaging here rather then just what the exception.getMessage() returns, for example, return what we have in the logger.error in the line above